examples/request-coalescing.ts

```diff
@@ -0,0 +1,431 @@
+    ).map(_ => this.launchAndRunWorkers(momentoWithCoalescing, loadGenContext));
+
+    await Promise.all(asyncGetSetResultsWithRequestCoalescer);
+
+    // We're done, stop showing stats.
+    clearInterval(logStatsIntervalId);
+
+    // Show the stats one last time at the end.
+    logStats(loadGenContext, this.logger, this.options.maxRequestsPerSecond);
+    logCoalescingStats(requestCoalescerContext, this.logger);
+
+    this.logger.info('DONE!');
+    // wait a few millis to allow the logger to finish flushing
+    await delay(500);
+  }
+
+  private async launchAndRunWorkers(
+    client: GetAndSetOnlyClient,
+    loadGenContext: BasicLoadGenContext
+  ): Promise<void> {
+    let finished = false;
+    const finish = () => (finished = true);
+    setTimeout(finish, this.options.totalSecondsToRun * 1000);
+
+    let i = 1;
+    for (;;) {
+      await this.issueAsyncSetGet(client, loadGenContext);
+      if (finished) {
+        return;
+      }
+      i++;
+    }
+  }
+
+  private async issueAsyncSetGet(
+    client: GetAndSetOnlyClient,
+    loadGenContext: BasicLoadGenContext
+  ): Promise<void> {
+    const cacheKey = this.getRandomCacheKey();
+
+    const setStartTime = process.hrtime();
+    const result = await this.executeRequestAndUpdateContextCounts(
+      loadGenContext,
+      () => client.set(this.cacheName, cacheKey, this.cacheValue)
+    );
+    if (result !== undefined) {
+      const setDuration = getElapsedMillis(setStartTime);
+      loadGenContext.setLatencies.recordValue(setDuration);
+      if (setDuration < this.delayMillisBetweenRequests) {
+        const delayMs = this.delayMillisBetweenRequests - setDuration;
+        this.logger.trace(`delaying: ${delayMs}`);
+        await delay(delayMs);
+      }
+    }
+
+    const getStartTime = process.hrtime();
+    const getResult = await this.executeRequestAndUpdateContextCounts(
+      loadGenContext,
+      () => client.get(this.cacheName, cacheKey)
+    );
+
+    if (getResult !== undefined) {
+      const getDuration = getElapsedMillis(getStartTime);
+      loadGenContext.getLatencies.recordValue(getDuration);
+      if (getDuration < this.delayMillisBetweenRequests) {
+        const delayMs = this.delayMillisBetweenRequests - getDuration;
+        this.logger.trace(`delaying: ${delayMs}`);
+        await delay(delayMs);
+      }
+    }
+  }
+
+  private getRandomCacheKey(): string {
+    return cacheKeys[Math.floor(Math.random() * cacheKeys.length)];
+  }
+
+  private initiateLoadGenContext(): BasicLoadGenContext {
+    const loadGenContext: BasicLoadGenContext = {
+      startTime: process.hrtime(),
+      getLatencies: hdr.build(),
+      setLatencies: hdr.build(),
+      globalRequestCount: 0,
+      globalSuccessCount: 0,
+      globalUnavailableCount: 0,
+      globalDeadlineExceededCount: 0,
+      globalResourceExhaustedCount: 0,
+      globalRstStreamCount: 0,
+    };
+    return loadGenContext;
+  }
+  private async executeRequestAndUpdateContextCounts<T>(
```

Comment:
if this function and the one below it are identical to the ones in load-gen, I think these would be reasonable to pull out into your utils classes.